### ISSUE TITLE
Extremely Low GPU Usage

### ISSUE THREAD

**Issue by Niknokinater** [IrisShaders/Iris#1304]
### What happened?

Entities & block entities **significantly** affect how much GPU resource is even used.

An underlying issue appears to be in the vanilla game, but the vanilla game barely uses any GPU in the first place, which is what I'm poking at here.

If anything more GPU-side can be done, that would be great.

### Screenshots

![iris1](https://user-images.githubusercontent.com/71681570/156867113-3d3c299e-f019-47f6-8893-d6526c5516f1.jpg)
![iris2](https://user-images.githubusercontent.com/71681570/156867149-2aa801ba-4f0d-4584-b400-6d39156e17cd.jpg)
![iris3](https://user-images.githubusercontent.com/71681570/156867260-e6b0b502-bcd6-4014-854b-d25bb796459a.jpg)
![iris4](https://user-images.githubusercontent.com/71681570/156867265-c7ef685d-4cd1-4d54-9f49-7cbfb51b3e2e.jpg)

### Relevant log output

```shell
[22:02:48] [Render thread/INFO]: Applied 0 biome modifications to 0 of 61 new biomes in 2.254 ms
[22:02:48] [Server thread/INFO]: Starting integrated minecraft server version 1.18.2
[22:02:48] [Server thread/INFO]: Generating keypair
[22:02:48] [Server thread/INFO]: Preparing start region for dimension minecraft:overworld
[22:02:49] [Render thread/INFO]: Preparing spawn area: 0%
[22:02:49] [Render thread/INFO]: Preparing spawn area: 0%
[22:02:49] [Render thread/INFO]: Preparing spawn area: 0%
[22:02:49] [Render thread/INFO]: Preparing spawn area: 0%
[22:02:50] [Render thread/INFO]: Preparing spawn area: 0%
[22:02:50] [Render thread/INFO]: Preparing spawn area: 1%
[22:02:51] [Render thread/INFO]: Preparing spawn area: 2%
[22:02:51] [Render thread/INFO]: Preparing spawn area: 22%
[22:02:52] [Render thread/INFO]: Preparing spawn area: 24%
[22:02:52] [Render thread/INFO]: Preparing spawn area: 31%
[22:02:53] [Render thread/INFO]: Preparing spawn area: 79%
[22:02:54] [Render thread/INFO]: Time elapsed: 6290 ms
[22:02:54] [Server thread/INFO]: Changing simulation distance to 5, from 0
[22:02:56] [Server thread/INFO]: Niknokinater[local:E:84f60277] logged in with entity id 377 at (-22.681239391781418, 72.0, -98.45125934929364)
[22:02:56] [Server thread/INFO]: Niknokinater joined the game
[22:02:56] [Render thread/INFO]: Started 3 worker threads
[22:02:57] [Render thread/INFO]: Loaded 727 advancements
[22:02:57] [Render thread/WARN]: Creating pipeline for dimension OVERWORLD
[22:02:57] [Render thread/WARN]: [deferred1] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [deferred1] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [deferred1] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [deferred1] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [deferred1] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [deferred1] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [composite] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [composite] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [composite] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [composite] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [composite] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [composite] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [composite1] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [composite1] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [composite1] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [composite1] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [composite1] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [composite1] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [Triforce Patcher] gl_FragColor is not supported yet, please use gl_FragData! Assuming that the shaderpack author intended to use gl_FragData[0]...
[22:02:57] [Render thread/WARN]: [basic] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [basic] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [basic] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [basic] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [basic] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [basic] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [basic_color] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [basic_color] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [basic_color] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [basic_color] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [basic_color] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [basic_color] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [textured] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [textured] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [textured] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [textured] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [textured] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [textured] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [textured_color] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [textured_color] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [textured_color] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [textured_color] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [textured_color] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [textured_color] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [sky_basic] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [sky_basic] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [sky_basic] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [sky_basic] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [sky_basic] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [sky_basic] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [sky_basic_color] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [sky_basic_color] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [sky_basic_color] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [sky_basic_color] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [sky_basic_color] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [sky_basic_color] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [clouds] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [clouds] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [clouds] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [clouds] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [clouds] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [clouds] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [terrain_solid] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [terrain_solid] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [terrain_solid] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [terrain_solid] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [terrain_solid] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [terrain_solid] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [terrain_cutout] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [terrain_cutout] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [terrain_cutout] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [terrain_cutout] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [terrain_cutout] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [terrain_cutout] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [terrain_cutout_mipped] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [terrain_cutout_mipped] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [terrain_cutout_mipped] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [terrain_cutout_mipped] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [terrain_cutout_mipped] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [terrain_cutout_mipped] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [terrain_translucent] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [terrain_translucent] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [terrain_translucent] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [terrain_translucent] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [terrain_translucent] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [terrain_translucent] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [entities_solid] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [entities_solid] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [entities_solid] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [entities_solid] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [entities_solid] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [entities_solid] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [entities_solid_diffuse] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [entities_solid_diffuse] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [entities_solid_diffuse] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [entities_solid_diffuse] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [entities_solid_diffuse] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [entities_solid_diffuse] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [entities_solid_bright] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [entities_solid_bright] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [entities_solid_bright] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [entities_solid_bright] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [entities_solid_bright] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [entities_solid_bright] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [entities_cutout] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [entities_cutout] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [entities_cutout] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [entities_cutout] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [entities_cutout] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [entities_cutout] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [entities_cutout_diffuse] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [entities_cutout_diffuse] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [entities_cutout_diffuse] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [entities_cutout_diffuse] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [entities_cutout_diffuse] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [entities_cutout_diffuse] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [hand_cutout] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [hand_cutout] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [hand_cutout] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [hand_cutout] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [hand_cutout] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [hand_cutout] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [hand_cutout_bright] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [hand_cutout_bright] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [hand_cutout_bright] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [hand_cutout_bright] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [hand_cutout_bright] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [hand_cutout_bright] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [hand_cutout_diffuse] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [hand_cutout_diffuse] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [hand_cutout_diffuse] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [hand_cutout_diffuse] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [hand_cutout_diffuse] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [hand_cutout_diffuse] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [hand_translucent] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [hand_translucent] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [hand_translucent] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [hand_translucent] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [hand_translucent] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [hand_translucent] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [hand_water_bright] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [hand_water_bright] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [hand_water_bright] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [hand_water_bright] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [hand_water_bright] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [hand_water_bright] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [hand_water_diffuse] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [hand_water_diffuse] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [hand_water_diffuse] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [hand_water_diffuse] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [hand_water_diffuse] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [hand_water_diffuse] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [lightning] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [lightning] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [lightning] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [lightning] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [lightning] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [lightning] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [leash] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [leash] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [leash] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [leash] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [leash] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [leash] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [particles] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [particles] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [particles] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [particles] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [particles] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [particles] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [weather] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [weather] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [weather] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [weather] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [weather] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [weather] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [text] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [text] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [text] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [text] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [text] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [text] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [text_intensity] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [text_intensity] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [text_intensity] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [text_intensity] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [text_intensity] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [text_intensity] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [block_entity] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [block_entity] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [block_entity] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [block_entity] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [block_entity] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [block_entity] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [block_entity_bright] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [block_entity_bright] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [block_entity_bright] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [block_entity_bright] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [block_entity_bright] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [block_entity_bright] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [block_entity_diffuse] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [block_entity_diffuse] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [block_entity_diffuse] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [block_entity_diffuse] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [block_entity_diffuse] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [block_entity_diffuse] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [lines] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [lines] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [lines] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [lines] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [lines] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [lines] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/INFO]: Stopping worker threads
[22:02:57] [Render thread/INFO]: Started 3 worker threads
[22:02:57] [Render thread/WARN]: [<sodium shaders>] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [<sodium shaders>] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [<sodium shaders>] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [<sodium shaders>] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [<sodium shaders>] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [<sodium shaders>] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [<sodium shaders>] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [<sodium shaders>] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [<sodium shaders>] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [<sodium shaders>] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [<sodium shaders>] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [<sodium shaders>] Unsupported uniform: float isSwamp
[22:02:57] [Render thread/WARN]: [<sodium shaders>] Unsupported uniform: float isCold
[22:02:57] [Render thread/WARN]: [<sodium shaders>] Unsupported uniform: float isDesert
[22:02:57] [Render thread/WARN]: [<sodium shaders>] Unsupported uniform: float isMesa
[22:02:57] [Render thread/WARN]: [<sodium shaders>] Unsupported uniform: float isMushroom
[22:02:57] [Render thread/WARN]: [<sodium shaders>] Unsupported uniform: float isSavanna
[22:02:57] [Render thread/WARN]: [<sodium shaders>] Unsupported uniform: float isSwamp
[22:03:18] [Server thread/INFO]: Saving and pausing game...
[22:03:18] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:overworld
[22:03:18] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:the_end
[22:03:18] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:the_nether
[22:03:24] [Render thread/INFO]: OpenGL debug message: id=1282, source=API, type=ERROR, severity=HIGH, message='GL_INVALID_OPERATION error generated. Function glPushAttrib is deprecated and not available in preview contexts.'
[22:03:24] [Render thread/INFO]: OpenGL debug message: id=1282, source=API, type=ERROR, severity=HIGH, message='GL_INVALID_OPERATION error generated. Function glPopAttrib is deprecated and not available in preview contexts.'
[22:03:46] [Server thread/INFO]: Saving and pausing game...
[22:03:46] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:overworld
[22:03:46] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:the_end
[22:03:46] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:the_nether
[22:05:20] [Render thread/INFO]: Flushed changes to Minecraft configuration
[22:05:56] [Render thread/INFO]: Flushed changes to Minecraft configuration
[22:06:09] [Server thread/INFO]: Saving and pausing game...
[22:06:09] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:overworld
[22:06:09] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:the_end
[22:06:09] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:the_nether
[22:07:22] [Server thread/INFO]: Saving and pausing game...
[22:07:22] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:overworld
[22:07:22] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:the_end
[22:07:22] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:the_nether
[22:07:49] [Server thread/INFO]: Saving and pausing game...
[22:07:49] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:overworld
[22:07:49] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:the_end
[22:07:49] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:the_nether
[22:10:19] [Server thread/INFO]: Saving and pausing game...
[22:10:19] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:overworld
[22:10:19] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:the_end
[22:10:19] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:the_nether
[22:14:29] [Server thread/INFO]: Saving and pausing game...
[22:14:29] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:overworld
[22:14:29] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:the_end
[22:14:29] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:the_nether
[22:14:49] [Server thread/INFO]: Saving and pausing game...
[22:14:49] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:overworld
[22:14:49] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:the_end
[22:14:49] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:the_nether
[22:18:15] [Server thread/INFO]: Niknokinater lost connection: Disconnected
[22:18:15] [Server thread/INFO]: Niknokinater left the game
[22:18:15] [Server thread/INFO]: Stopping singleplayer server as player logged out
[22:18:15] [Server thread/INFO]: Stopping server
[22:18:15] [Server thread/INFO]: Saving players
[22:18:15] [Server thread/INFO]: Saving worlds
[22:18:16] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:overworld
[22:18:16] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:the_end
[22:18:16] [Server thread/INFO]: Saving chunks for level 'ServerLevel[Tenoof | 118_SURV, HARDCORE]'/minecraft:the_nether
[22:18:16] [Server thread/INFO]: ThreadedAnvilChunkStorage (Tenoof _ 118_SURV, HARDCORE): All chunks are saved
[22:18:16] [Server thread/INFO]: ThreadedAnvilChunkStorage (DIM1): All chunks are saved
[22:18:16] [Server thread/INFO]: ThreadedAnvilChunkStorage (DIM-1): All chunks are saved
[22:18:16] [Server thread/INFO]: ThreadedAnvilChunkStorage: All dimensions are saved
[22:18:16] [Render thread/INFO]: Stopping worker threads
```

### Minecraft Version

Minecraft 1.18.2

### Iris Version

iris-mc1.18.2-1.2.1-rc2-0cc372f0-dirty.jar

### Sodium Version

sodium-fabric-mc1.18.2-0.4.1+build.15.jar

### Operating System

Windows 10

### What is your GPU?

NVidia GeForce RTX 3070Ti (497.29)

### Java Version

Java 17

### Additional context

For reference, F3 debug screen always halves framerate.
FPS capped at 120 in all images.
Additionally, Sodium standalone [uncapped] framerates are ~460 looking at sky, ~150 looking at structure.
The structure is filled with chests.
CPU is i7-7700, typically clocking 4.1GHz across all 8 threads under load.

**Comment by superwizzman** [comment]
Are-you sure it's a bug? You only use 2go of RAM. Maybe try with 16?

**Comment by Niknokinater** [comment]
The amount of allocated RAM does not appear to make a difference. I've tried 2, 8, 16, etc.
I'm certain that it's a vanilla limitation that Sodium / Iris isn't considering, should it even be possible to work around.

**Comment by Inakei23189** [comment]
I have the same issue.
These below are the same world, place, mods(with different versions) and exactly the same settings.
But low fps at 1.18.2 with IrisShaders.

1.18.1
iris-mc1.18.1-1.2.0-pre.jar
sodium-fabric-mc1.18.1-0.4.0-alpha6+build.14.jar
Stable 60fps with GREEN letters of "[Iris] Version: 1.2.0-pre"
![Minecraft_ 1 18 1_ 2022_03_09 19_45_26](https://user-images.githubusercontent.com/101049001/157663532-af803397-c349-4a40-a6e0-0666ee461061.png)

1.18.2
iris-mc1.18.2-1.2.1-rc2-0cc372f0-dirty.jar
sodium-fabric-mc1.18.2-0.4.1+build.15.jar
Around 30fps with RED letters of "[Iris] Version: 1.2.1-rc2-0cc372f0-dirty"
![Minecraft_ 1 18 2_ 2022_03_09 19_44_03](https://user-images.githubusercontent.com/101049001/157663662-f538c9a8-a0b4-42ee-a6a2-c68ddc4b247a.png)

**Comment by coderbot16** [comment]
This should be mostly resolved (back up to 1.2.0 levels) in 1.2.2.

**Comment by Niknokinater** [comment]
I was not expecting such a boost. Whilst I'm only reaching the 40s now at the same location, the fact that nearly double the fps came about is actually pretty amazing.

Thank you very much ! Here's to hoping underlying issues in the base game + Sodium are optimized further.
